Lib/test/test_typing.py

```diff
@@ -5793,6 +5793,14 @@ def test_paramspec_in_nested_generics(self):
         self.assertEqual(G2[[int, str], float], list[C])
         self.assertEqual(G3[[int, str], float], list[C] | int)
 
+    def test_paramspec_gets_copied(self):
+        # bpo-46581
+        P = ParamSpec('P')
+        original = Callable[P, int]
+        self.assertEqual(original.__parameters__, (P,))
+        copied = original[P]
+        self.assertEqual(original.__parameters__, copied.__parameters__)
```

Comment:
Could you add some more complicated test cases? Ideas:
- The original has a `Concatenate[int, P]` in it (`original = Callable[Concatenate[int, P], int]; original[P].__paramaters__ == (P,)`)
- Instead of substituting a `P` in, substitute in a `Concatenate` or something like `[int, T]`, and then the `T` should be in `__parameters__`.